animations/transition_widgets.md

```diff
@@ -0,0 +1,200 @@
+---
+layout: page
+title: "Flutter transition animations"
+permalink: /animations/transitions_widgets.html
+---
+
+* TOC Placeholder
+{:toc}
+
+Transition animations implement the abstract class `AnimatedWidget`. The `AnimatedWidget` class allows you to separate the widget code from the animation code in the `setState()` call. The `AnimatedWidget` class doesn’t need to maintain a `State` object to hold the animation. Use the  `AnimatedWidget` helper class (instead of `addListener()` and `setState()`) to create a reusable transition animation widget.
```

Comment:
Can we talk about how to create your own transition widgets _after_ talking about how to use the transition widgets provided by Flutter? Most people probably don't need to create their own transition widgets.